aiohttp/web_urldispatcher.py

```diff
@@ -535,14 +535,11 @@ def __init__(
     ) -> None:
         super().__init__(prefix, name=name)
         try:
-            directory = Path(directory)
-            if str(directory).startswith("~"):
-                directory = Path(os.path.expanduser(str(directory)))
-            directory = directory.resolve()
-            if not directory.is_dir():
-                raise ValueError("Not a directory")
-        except (FileNotFoundError, ValueError) as error:
+            directory = Path(directory).expanduser().resolve(True)
```

Comment:
I'd suggest using a named parameter to save needing to look up the docs.
```suggestion
            directory = Path(directory).expanduser().resolve(strict=True)
```